### PR TITLE
Provide `zero-copy` cargo feature

### PR DESCRIPTION
**Pull request by temeddix**
## Changes

This PR started from #1222

From `allo-isolate` 0.1.15, when enabling `zero-copy` cargo feature, `Vec<u8>` and its friends will be zero-copy by default. As flutter_rust_bridge is a thin wrapper around Dart VM, this convenience will benefit many developers using this library. We don't need to use `ZeroCopyBuffer` around our code anymore.

Docs are updated regarding this trait implementation.
https://github.com/sunshine-protocol/allo-isolate/blob/6956e217e4d3c09fac10778bfa1b89176d3cd285/src/into_dart.rs#LL232C1-L244C1

Please be aware that `allo-isolate` 0.1.15 is not released yet, but it will be soon. I made a PR at `allo-isolate` repository and it will make it into the new version in a few days.
https://github.com/sunshine-protocol/allo-isolate/pull/39

## Checklist

- [x] An issue to be fixed by this PR is listed above.
- [x] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
- [x] The code generator is run and the code is formatted (via `just precommit`).
- [x] If this PR adds/changes features, documentations (in the `./book` folder) are updated.
- [x] CI is passing.

## Remark for PR creator

- Justfile is a task runner for the command line interface that allows you to run shell commands from a file named `justfile` in your project directory. You can use Justfile after [installing it](https://github.com/casey/just). Note that commands written in `justfile` of this repository are expected to be run in `bash`, not `cmd` or `powershell`. Running `just ...` commands in `cmd` or `powershell` will produce errors as the syntax is not compatible. On Windows, you can use `git bash` if you have Git installed.
- If fzyzcjy does not reply for a few days, maybe he just did not see it, so please ping him.

### CODE REVIEW

**Comment by fzyzcjy** [comment]
Good job! I checked https://github.com/sunshine-protocol/allo-isolate/issues/38#issuecomment-1561020261 and it seems reasonable to make it under a feature flag. So, shall we also expose this feature flag explicitly for end users?

**Comment by fzyzcjy** [comment]
No hurries! I also commented at https://github.com/sunshine-protocol/allo-isolate/issues/38#issuecomment-1564396101, and is curious which one we should choose as default.

**Comment by temeddix** [comment]
Hmm, then should I modify the PR _after_ the default is chosen?

File: Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ categories = ["development-tools::ffi"]
 
 [workspace.dependencies]
 allo-isolate = { version = "0.1.14-beta.3", features = ["anyhow"] }
+# allo-isolate = { version = "0.1.15", features = ["anyhow", "zero-copy"] }
```
[the remark this answers is not in the document]
Comment (temeddix) [reply Cargo.toml:22]: Oh, thank you very much for letting us know

**Comment by temeddix** [comment]
I didn't change much, but the CI is failing :(

**Comment by fzyzcjy** [comment]
MacOS CI error: https://github.com/fzyzcjy/flutter_rust_bridge/issues/1225

**Comment by fzyzcjy** [comment]
I do not have strong opinion whether to opt-in or opt-out. But for best backward compatibility maybe new features should be under a flag, so maybe let's opt-in firstly ;)

**Review by fzyzcjy — APPROVED** [review]
Good job! I will merge if there is no further changes

**Comment by temeddix** [comment]
Yeah, I also think that would be a reasonable decision 😀

**Pull request merged**